### PR TITLE
Teleporter

### PR DESCRIPTION
**Pull request by Zalgo239**
**What:**
Adds a teleporter app to the terminal.
Would work pretty well as a non-cheaty way to access teleportation, at least in the late game

**Implementation Details:**
Teleportation is done by a simple portal entity, spawned thanks to a new app in the terminal, any entity that walks inside the portal is instantly teleported to the requested destination 

**Outcome:**
+Added teleporter app and portal entity

**Additional info:**
Closing animation (opening is the same in reverse)
https://user-images.githubusercontent.com/67443118/160469833-d8f0c595-b13f-4e75-81f2-b9b4850b68cc.mp4

Full opening and closing, file is too large so I have to just link the discord
https://discord.com/channels/701354865217110096/701354865657643070/958081017065914468

### CODE REVIEW

File: src/main/java/gregtech/common/entities/PortalEntity.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                TeleportHandler.teleport(entity, TargetDim, teleporter,TargetX + entity.getLookVec().x, TargetY, TargetZ + entity.getLookVec().z);
+            }
+        }
+        --timeToDespawn;
+    }
+
+    @Override
+    public void setRotation(float yaw, float pitch){
+        super.setRotation(yaw, pitch);
+    }
```
Comment (Rongmario) [inline src/main/java/gregtech/common/entities/PortalEntity.java:85]: Unnecessary?

File: src/main/java/gregtech/common/terminal/app/teleport/TeleportApp.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.chunk.Chunk;
+
+public class TeleportApp extends AbstractApplication {
+
+    private int CoordinateX = 0;
+    private int CoordinateY = 1;
+    private int CoordinateZ = 0;
+
+    private int Dimension = 0;
```
Comment (Rongmario) [inline src/main/java/gregtech/common/terminal/app/teleport/TeleportApp.java:21]: Conventions

Comment (Zalgo239) [reply src/main/java/gregtech/common/terminal/app/teleport/TeleportApp.java:21]: wdym?

Comment (brachy84) [reply src/main/java/gregtech/common/terminal/app/teleport/TeleportApp.java:21]: We use lower case var names is what rong meant i think.

Comment (Zalgo239) [reply src/main/java/gregtech/common/terminal/app/teleport/TeleportApp.java:21]: Alright, will change that

**Review by Rongmario — APPROVED** [review]

**Comment by Quarri6343** [comment]
I can't see this app is functional...
There is no description in the app so it is very hard to understand what parameter in GUI does what
When I use it it just keep teleporting me to the y=-1

**Comment by brachy84** [comment]
Yeah, it definitly needs some description, but i assume it is x, y, z, dimension

**Comment by Quarri6343** [comment]
I was thinking same but I found that was wrong
it has default value 0, 1, 0 but it doesn't bring me to that coordinate
I'll look into the code if I have time

**Review by brachy84 — APPROVED** [review]
No issues found. I enhanced the ui a little bit. And also made it so it saves the last pos typed in.

**Pull request merged**